fontc/src/work.rs

```diff
@@ -0,0 +1,115 @@
+//! Helps fontc manage workloads that span FE and BE.
+//!
+//! Basically enums that can be a FeWhatever or a BeWhatever.
+
+use std::{collections::HashSet, fmt::Display};
+
+use fontbe::{
+    error::Error as BeError,
+    orchestration::{AnyWorkId, BeWork, Context as BeContext},
+};
+use fontir::{
+    error::WorkError as FeError,
+    orchestration::{Context as FeContext, IrWork},
+};
+
+#[derive(Debug)]
+pub enum AnyWorkError {
+    Fe(FeError),
+    Be(BeError),
+}
+
+impl From<BeError> for AnyWorkError {
+    fn from(e: BeError) -> Self {
+        AnyWorkError::Be(e)
+    }
+}
+
+impl From<FeError> for AnyWorkError {
+    fn from(e: FeError) -> Self {
+        AnyWorkError::Fe(e)
+    }
+}
+
+impl Display for AnyWorkError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AnyWorkError::Be(e) => e.fmt(f),
+            AnyWorkError::Fe(e) => e.fmt(f),
+        }
+    }
+}
+
+// Work of any type, FE, BE, ... some future pass, w/e
+pub enum AnyWork {
+    Fe(Box<IrWork>),
+    Be(Box<BeWork>),
+}
+
+impl From<Box<IrWork>> for AnyWork {
+    fn from(work: Box<IrWork>) -> Self {
+        AnyWork::Fe(work)
+    }
+}
+
+impl From<Box<BeWork>> for AnyWork {
+    fn from(work: Box<BeWork>) -> Self {
+        AnyWork::Be(work)
+    }
+}
+
+impl AnyWork {
+    pub fn exec(&self, context: AnyContext) -> Result<(), AnyWorkError> {
+        match self {
+            AnyWork::Be(work) => work.exec(context.unwrap_be()).map_err(|e| e.into()),
+            AnyWork::Fe(work) => work.exec(context.unwrap_fe()).map_err(|e| e.into()),
+        }
+    }
+}
+
+pub enum AnyContext {
+    Fe(FeContext),
+    Be(BeContext),
+}
+
+impl AnyContext {
+    pub fn for_work(
+        fe_root: &FeContext,
+        be_root: &BeContext,
+        id: &AnyWorkId,
+        happens_after: HashSet<AnyWorkId>,
```

Comment:
I suspect we would find some joy in a `WorkSet` type.

Reply:
An id set? Why? - we only use it for very hashset'ish things currently.

Reply:
I didn't make that change as I wasn't quite sure why. Happy to make it in a follow-on PR if you wish.